examples/hybrid_amazon_linux/main.tf

```diff
@@ -0,0 +1,345 @@
+    "KONG_ANONYMOUS_REPORTS" = "off"
+    "KONG_PORTAL"            = "on"
+    "KONG_VITALS"            = "on"
+    "KONG_AUDIT_LOG"         = "on"
+    "KONG_LOG_LEVEL"         = "debug"
+  }
+
+  kong_data_plane_config = {
+    "KONG_ROLE"              = "data_plane"
+    "KONG_DATABASE"          = "off"
+    "KONG_LOG_LEVEL"         = "debug"
+    "KONG_ANONYMOUS_REPORTS" = "off"
+  }
+
+  kong_hybrid_conf = {
+    server_name  = ""
+    cluster_cert = tls_locally_signed_cert.cert.cert_pem
+    cluster_key  = tls_private_key.cert.private_key_pem
+    mtls         = "shared"
+    ca_cert      = ""
+    endpoint     = aws_lb.internal.dns_name
+  }
+}
+
+module "create_kong_cp" {
+  source = "../../"
+
+  instance_type             = var.instance_type
+  vpc_id                    = aws_vpc.vpc.id
+  ami_id                    = data.aws_ami.amazon_linux_2.id
+  ami_operating_system      = "amazon-linux"
+  ce_pkg                    = "kong-enterprise-edition-2.3.2.0.rhel7.noarch.rpm" # there is no specific CE binary on bintray
+  key_name                  = var.key_name
+  region                    = var.region
+  vpc_cidr_block            = aws_vpc.vpc.cidr_block
+  iam_instance_profile_name = aws_iam_instance_profile.kong.name
+
+  ee_creds_ssm_param = {
+    license          = aws_ssm_parameter.ee-license.name
+    bintray_username = aws_ssm_parameter.ee_bintray_username.name
+    bintray_password = aws_ssm_parameter.ee_bintray_password.name
+    admin_token      = aws_ssm_parameter.ee-admin-token.name
+  }
+
+  asg_desired_capacity = var.asg_desired_capacity
+  asg_max_size         = var.asg_max_size
+  asg_min_size         = var.asg_min_size
+
+  proxy_config = {
+    http_proxy  = "http://${aws_instance.external_proxy.private_ip}:3128"
+    https_proxy = "http://${aws_instance.external_proxy.private_ip}:3128"
+    no_proxy    = "localhost,169.254.169.254,127.0.0.1"
+  }
+
+  rules_with_source_cidr_blocks = var.rules_with_source_cidr_blocks
+
+  postgres_config = {
+    master_user     = var.postgres_master_user
+    master_password = random_string.master_password.result
+  }
+
+  postgres_host = aws_instance.external_postgres.private_ip
+
+  kong_database_config = {
+    user     = var.kong_database_user
+    name     = var.kong_database_name
+    password = var.kong_database_password
+  }
+
+  target_group_arns = local.target_group_cp
+
+  skip_rds_creation = true
+  kong_config       = local.kong_control_plane_config
+  kong_hybrid_conf  = local.kong_hybrid_conf
+
+  environment = local.environment
+  service     = var.service
+  description = var.description
+  tags        = var.tags
+}
+
+module "create_kong_dp" {
+  source = "../../"
+
+  instance_type  = var.instance_type
+  vpc_id         = aws_vpc.vpc.id
+  ami_id                    = data.aws_ami.amazon_linux_2.id
+  ami_operating_system = "amazon-linux"
+  ce_pkg = "kong-enterprise-edition-2.3.2.0.rhel7.noarch.rpm" # there is no specific CE binary on bintray
```

Comment:
Formatting is slightly off. Down to personal preference but could clean up if wanted